vaadin-spring/src/test/java/com/vaadin/flow/spring/SpringHelperTest.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright 2000-2017 Vaadin Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.vaadin.flow.spring;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.junit.Assert;
+import org.junit.Test;
+import org.springframework.core.env.AbstractEnvironment;
+import org.springframework.core.env.Environment;
+import org.springframework.core.env.MapPropertySource;
+import org.springframework.core.env.MutablePropertySources;
+import org.springframework.core.env.PropertySource;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class SpringHelperTest {
+
+    @Test
+    public void camelCase_normalized_matches() {
+        String normalized = SpringHelper
+                .normalizePropertyName("vaadin.urlMapping");
+        Assert.assertEquals("vaadinurlmapping", normalized);
```

Comment:
This is exactly my concern. 
Now I can use `vaadinurlmapping` for `vaadin.url-mapping` .
So : 
- `vaadin` prefix is not a prefix anymore
- the `urlmapping` is just the same as  `url-mapping`.

Reply:
As commented before, I think this is correct. Well, I would better say that it works / is acceptable.

I would prefer to use the read the values directly from `VaadinConfigurationProperties`, so all this code would be unnecessary. In fact, this is what I did at the begining, until I discovered the MVC part of the integration and I realized that, yes or yes, I needed to rely on properties.

Once that we accept that we must rely on properties, and that spring accepts that relaxed naming, we must be able to find the property whatever name it is given. Hence all this code.

Anyway, as said before, I think it is feasible to say that we will not use `vaadinurlmapping` and `vaadin.url-mapping` for different properties.

Reply:
It doesn't matter whether we use `vaadinurlmapping` or not: this is not a Vaadin property at all.
The property  `vaadinurlmapping`  may be accidentally used by the developer and it should not match `vaadin.url-mapping`.

Again: the current set of properties and the code are not final. So it means that the code is evolving.
At some point the same code from `SpringHelper` can be used for some other property and then the mistake won't be so obvious.

I don't see :
- reasons to have this as a utility method
- it looks like it's used only in one specific case and it's easier to make normalization/converstation directly as a private method specifically for this case.